src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemorySegment.java

```diff
@@ -184,7 +184,8 @@
      *
      * @param elementLayout the layout to be used for splitting.
      * @return the element spliterator for this segment
-     * @throws IllegalArgumentException if this segment size is not a multiple of the size of {@code elementLayout}.
+     * @throws IllegalArgumentException if this segment size is not a multiple of the size of {@code elementLayout},
```

Comment:
"... if the elementLayout size is zero, or the segment size modulo the elementLayout size is greater than zero" ?